locations/spiders/ingles.py

```diff
@@ -0,0 +1,78 @@
+# -*- coding: utf-8
+import scrapy
+import re
+
+from locations.items import GeojsonPointItem
+from locations.hours import OpeningHours
+
+URL = 'https://www.ingles-markets.com/storelocate/storelocator.php?address='
+
+STORE_STATES = ["Alabama", "Georgia", "North%20Carolina", "South%20Carolina", "Tennessee", "Virginia"]
+
+DAYS = ["Mo", "Tu", "We", "Th", "Fr", "Sa", "Su"]
+
+class ingles(scrapy.Spider):
+    name = "ingles"
+    item_attributes = { 'brand': "Ingles" }
+    allowed_domains = ["www.ingles-markets.com"]
+
+    def start_requests(self):
+        for state in STORE_STATES:
+            url = ''.join([URL, state])
+            yield scrapy.Request(url, callback=self.parse)
+
+    def parse_hours(self, hours):
+        opening_hours = OpeningHours()
+
+        for day in DAYS:
+            open_time, close_time = hours.split('to')
+            opening_hours.add_range(day=day, open_time=("".join(open_time).strip()), close_time=("".join(close_time).strip()), time_format="%H:%M%p")
+
+        return opening_hours.as_opening_hours()
+
+    def parse_store(self, response):
+
+        properties = {
+            'ref': response.meta["ref"],
+            'name': response.meta["name"],
+            'addr_full': response.meta["addr_full"],
+            'city': response.meta["city"],
+            'state': response.meta["state"],
+            'postcode': re.search(r'(\d{5})',response.xpath("/html/body/fieldset/div[2]/span[2]/strong/text()").get()).group(),
+            'phone': response.xpath("/html/body/fieldset/div[2]/a/text()").get(),
+            'lat': response.meta["lat"],
+            'lon': response.meta["lon"],
+            'website': response.url,
+        }
+
+        hours = self.parse_hours(" ".join(response.xpath("/html/body/fieldset/div[2]/text()")[2].getall()).strip())
+        if hours:
+            properties["opening_hours"] = hours
+
+        yield GeojsonPointItem(**properties)
+
+    def parse(self, response):
+        for store in response.xpath('//markers/marker'):
+            ids = str(store.xpath('./@id').get()),
+            name = store.xpath('./@name').get()
+            addr = store.xpath('./@address').get()
+            city = store.xpath('./@city').get()
+            state = store.xpath('./@state').get()
+            lats = store.xpath('./@lat').get()
+            longs = store.xpath('./@lng').get()
+
+            for id in ids:
+                store_url = 'https://www.ingles-markets.com/storelocate/storeinfo.php?storenum=' + id
+                yield scrapy.Request(
+                    response.urljoin(store_url),
```

Comment:
No need to use `urljoin()` if you're going to specify a full URL.

```suggestion
                yield scrapy.Request(
                    'https://www.ingles-markets.com/storelocate/storeinfo.php?storenum=' + id,
```